Formatting changes, Add windows support

moritzfl opened the pull request:
Added windows support and changed formatting to be [PEP 8](https://www.python.org/dev/peps/pep-0008/) conform.
 
Example of usage can be found in this wiki (clone locally to see all files - github only shows *.md files otherwise):
https://github.com/KernelHaven/KernelHaven/wiki (for webbrowser)
https://github.com/KernelHaven/KernelHaven.wiki.git (for cloning via git)
Information can be found in the file [Readme.txt](https://raw.githubusercontent.com/wiki/KernelHaven/KernelHaven/Readme.txt) of the same repo

moritzfl committed:
Formatting changes, Add windows support

Changed formatting to be [PEP 8](https://www.python.org/dev/peps/pep-0008/) conform.

jobisoft commented:
Thanks for the PEP8 work. 

If we are going to support windows, don't we always have to use os.sep instead of "/"? What about line 170 and 171:
https://github.com/jobisoft/wikidoc/pull/2/files#diff-dbd2fef04c3ab52919ed37d8390dec47R170

I cannot test this without setting up a new VM, which I currently do not have the time for...

moritzfl commented:
The PEP work was mostly done by my IDE (IntelliJ Idea) 😄 

Windows generally should be POSIX compatible via [this](https://en.m.wikipedia.org/wiki/Microsoft_POSIX_subsystem). However - and this was the first time that I stumbled into it - it does not seem to be POSIX compatible for calling an executable through its file-path via command line.

Of course one could now simply pass ```.\wkhtmltox-windows\bin\wkhtmltopdf.exe``` as the first argument but usually I would not assume a problem passing ```./wkhtmltox-windows/bin/wkhtmltopdf.exe``` as such a path definition works everywhere else. It acually took me quite a bit of time to find out what the problem was compared to how simple the fix was.

So while stuff like this works (you can even mix / and \ in a file-path and Windows still takes it):
```
  C:\Users\moritz> cd /Users
  C:\Users>
```

this does not (=without replacing / with \\):
*Note: I Inserted a print to show the cmd*
```
C:\Users\moritz\Desktop\KernelHaven.wiki>python wikidoc.py ./wkhtmltox-windows/bin/wkhtmltopdf.exe ./
Using _Sidebar.md for ordering of md-files
./wkhtmltox-windows/bin/wkhtmltopdf.exe --page-size A4 --margin-top 2cm --margin-left 2cm --margin-bottom 2cm --margin-right 2cm --footer-font-name Verdana --footer-font-size 12 --footer-spacing 10 --footer-right [page] cover wikidoc_cover.html toc --xsl-style-sheet wikidoc_toc.xsl wikidoc.html ./documentation.pdf
Der Befehl "." ist entweder falsch geschrieben oder konnte nicht gefunden werden.

C:\Users\moritz\Desktop\KernelHaven.wiki>python wikidoc.py wkhtmltox-windows/bin/wkhtmltopdf.exe ./
Using _Sidebar.md for ordering of md-files
wkhtmltox-windows/bin/wkhtmltopdf.exe --page-size A4 --margin-top 2cm --margin-left 2cm --margin-bottom 2cm --margin-right 2cm --footer-font-name Verdana --footer-font-size 12 --footer-spacing 10 --footer-right [page] cover wikidoc_cover.html toc --xsl-style-sheet wikidoc_toc.xsl wikidoc.html ./documentation.pdf
Der Befehl "wkhtmltox-windows" ist entweder falsch geschrieben oder konnte nicht gefunden werden.
```

After "fixing" just the path to the exe (=my commit), both of the commands used above work:

```
C:\Users\moritz\Desktop\KernelHaven.wiki>python wikidoc.py ./wkhtmltox-windows/bin/wkhtmltopdf.exe ./
Using _Sidebar.md for ordering of md-files
.\wkhtmltox-windows\bin\wkhtmltopdf.exe --page-size A4 --margin-top 2cm --margin-left 2cm --margin-bottom 2cm --margin-right 2cm --footer-font-name Verdana --footer-font-size 12 --footer-spacing 10 --footer-right [page] cover wikidoc_cover.html toc --xsl-style-sheet wikidoc_toc.xsl wikidoc.html ./documentation.pdf
Loading pages (1/6)
Counting pages (2/6)
Loading TOC (3/6)
Resolving links (4/6)
Loading headers and footers (5/6)
Printing pages (6/6)
Done
```

So in the end its Microsoft weirdness and only the path to the exe needs to be adjusted.

moritzfl committed:
Fixes for _Sidebar.md handling

Fixes an issue where _Sidebar.md was only found when located directly inside of the working directory from which wikidoc is called instead of looking into the wiki-folder.

Adds a check for entries in _Sidebar.md so that only entries get included in the filelist where a corresponding file exists (eg. : an entry like [Our Website](www.example.com) in  _Sidebar.md now gets ignored for the rendered document as it does not represent a markdown file)

jobisoft commented:
give me a few days, I am currently sick

moritzfl commented:
sure thing - no worries 👍
Get well soon!

moritzfl committed:
Fix duplicate wiki-path as prefix

jobisoft commented:
Thanks!
